role/auditor/rpc.go

```diff
@@ -0,0 +1,188 @@
+package auditor
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"net/rpc"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"time"
+	"validation-bot/module"
+
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+)
+
+type IClientRPC interface {
+	CallServer(ctx context.Context, input module.ValidationInput) (*module.ValidationResult, error)
+	Validate(
+		ctx context.Context,
+		stdout io.Reader,
+		input module.ValidationInput,
+	) (*module.ValidationResult, error)
+	GetTimeout() time.Duration
+}
+
+type ClientRPC struct {
+	log      zerolog.Logger
+	baseDir  string
+	Timeout  time.Duration
+	execPath string
+}
+
+type ClientConfig struct {
+	BaseDir  string
+	Timeout  time.Duration
+	ExecPath string
+}
+
+const (
+	scanPause = time.Millisecond * 200
+	scanLoops = 3
+)
+
+func NewClientRPC(config ClientConfig) *ClientRPC {
+	return &ClientRPC{
+		log:      log.With().Str("role", "rpc.client").Caller().Logger(),
+		baseDir:  config.BaseDir,
+		Timeout:  config.Timeout,
+		execPath: config.ExecPath,
+	}
+}
+
+func (r *ClientRPC) GetTimeout() time.Duration {
+	return r.Timeout
+}
+
+func (r *ClientRPC) CallServer(
+	ctx context.Context,
+	input module.ValidationInput,
+) (*module.ValidationResult, error) {
+	dir, err := os.MkdirTemp(r.baseDir, "validation_rpc")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create directory")
+	}
+
+	defer os.RemoveAll(dir)
+
+	absdir, err := filepath.Abs(dir)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get absolute path")
+	}
+
+	r.log.Info().Str("Absolute dir", absdir).Msg("using absolute path to tmp dir")
+	r.log.Info().Str("Exec Path", r.execPath).Msg("executing validation bot rpc from path")
+
+	// calls /path/to/ValidationBot/validation_bot validation-rpc
+	cmd := exec.CommandContext(ctx, dir, "validation-rpc")
```

Comment:
why not just run the module and get the results from the stdout from the cli and not use an rpc server?